comp/aggregator/demultiplexer/demultiplexer_mock.go

```diff
@@ -0,0 +1,18 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2023-present Datadog, Inc.
+
+//go:build test
+
+package demultiplexer
+
+import (
+	"github.com/DataDog/datadog-agent/pkg/aggregator"
+)
+
+func newMock(deps aggregator.TestDeps) Component {
+	return demultiplexer{
+		AgentDemultiplexer: aggregator.InitAndStartAgentDemultiplexerForTest(deps, aggregator.AgentDemultiplexerOptions{}, ""),
```

Comment:
Will there be a way to set these options? E.g. in our tests we don't want to wait for flushes, so previously we were creating an agg with `aggregator.InitTestAgentDemultiplexerWithFlushInterval(logger, 10*time.Millisecond)` to set the flush interval to 10ms instead of the default 15s; I've seen other tests that don't want flushes and so set the flush interval to an hour. How do we do that in a component-based test?

Reply:
When I opened the PR, I realised that this is not really a mock implementation. I opened the PR because I have other changes blocked by this PR. I am going to improve this part in a followup PR. This is tracked by `ASCII-811`.